### PR TITLE
[Test] Add CPU coverage for async invariant probes

### PR DESCRIPTION
**Pull request by mottopanikeiku**
## Motivation

`python/sglang/srt/utils/async_probe.py` provides environment-gated safety checks for NaN/Inf values, index ranges, page alignment, and optional in-place logit sanitization. These contracts previously had no direct unit coverage. Regressions in gate behavior, boundary semantics, empty-input handling, or sanitization could affect sampling and cache paths without a focused test identifying the source.

This is a focused slice of #20865.

## Modifications

- Add `test/registered/unit/utils/test_async_probe.py`.
- Add CPU-only coverage for:
  - disabled async assertions;
  - NaN and Inf detection;
  - inclusive closed-range boundaries;
  - lower-inclusive and upper-exclusive out-of-bounds checks;
  - aligned and misaligned page indices;
  - `None` and empty tensor handling;
  - enabled and disabled logit sanitization;
  - in-place replacement of NaN, positive Inf, and negative Inf;
  - hard-assert precedence over sanitization.
- Register the test in `base-a-test-cpu` with `est_time=45`, based on the measured 40.67-second cold-file run plus a small margin.
- Keep the CUDA-only `_AsyncNanWarner` pinned-memory path out of scope.

The tests do not launch a server, load model weights, access the network, or require an accelerator.

## Accuracy Tests

N/A. This is a test-only change and does not modify model outputs or runtime behavior.

## Speed Tests and Profiling

N/A. This is a CPU-only unit-test addition and does not modify inference performance.

## Validation

```text
$ python -m pytest test/registered/unit/utils/test_async_probe.py -v
collected 9 items

test_closed_range_includes_both_boundaries PASSED [ 11%]
test_disabled_assertions_are_noops PASSED [ 22%]
test_hard_assert_precedes_sanitization PASSED [ 33%]
test_nan_and_inf_detection PASSED [ 44%]
test_none_and_empty_tensors_are_ignored PASSED [ 55%]
test_oob_uses_inclusive_low_and_exclusive_high PASSED [ 66%]
test_page_alignment_and_skip_cases PASSED [ 77%]
test_sanitization_disabled_preserves_logits PASSED [ 88%]
test_sanitization_replaces_non_finite_logits_in_place PASSED [100%]

9 passed, 18 warnings, 6 subtests passed in 16.96s
```

The initial cold-file run took 40.67 seconds; that measurement is used for CI registration. The warnings came from existing accelerator availability checks, Torch deprecations, and the local pytest configuration; none originated from the added tests.

`pre-commit run --files test/registered/unit/utils/test_async_probe.py` passes all applicable hooks, including Python AST validation, isort, Ruff, Black, codespell, and registered-test CI validation.

## Independent review

A separate reviewer found no functional correctness issue: the tests exercise real `torch._assert_async` predicates and real in-place `torch.nan_to_num_`, restore environment patches, and correctly leave CUDA stream/pinned-memory timing out of scope. It flagged the original `est_time=2` as inconsistent with the measured cold-file runtime; the current commit corrects it to 45 seconds.

## Checklist

- [x] Format code according to the SGLang contribution guide.
- [x] Add unit tests according to the contribution guide.
- [ ] Update documentation. N/A: no public behavior or interface changed.
- [x] Follow the SGLang code style guidance.

---
### CI States

Latest PR Test (Base): :x: [Run #30140111496](https://github.com/sgl-project/sglang/actions/runs/30140111496)
Latest PR Test (Extra): :x: [Run #30140111428](https://github.com/sgl-project/sglang/actions/runs/30140111428)

### CODE REVIEW

**Comment by mottopanikeiku** [comment]
Hi @ispobock, this PR adds the focused CPU-only async_probe.py coverage claimed under #20865. The focused test and 
 pre-commit checks pass locally.                                                                                      
                                                                                                                      
   The initial workflows stopped at the shared gate because the PR does not yet have the run-ci label, and my         
 /tag-run-ci-label command was rejected because I do not have CI-label permission.                                    
                                                                                                                      
   Could you or another authorized maintainer please run /tag-and-rerun-ci for the current commit? Thank you!

**Comment by mottopanikeiku** [comment]
Independent review found no functional test issue, but correctly flagged that `est_time=2` understated the PR's measured 40.67-second cold-file runtime. Commit `7adfdc12e` updates registration to `est_time=45`. The focused suite still passes (9 tests, 6 subtests, 16.96s on the latest warm run), and pre-commit passes. The PR body is also reformatted and updated with the current evidence. The `/tag-and-rerun-ci` request applies to this commit.

**Comment by mottopanikeiku** [comment]
GitHub's PR object stopped tracking the fork branch after its main-branch sync: the branch ref is at 7adfdc12e, while the PR remains pinned to e6ae0e81a. Closing and immediately reopening to refresh the head association; no code is being dropped.

**Comment by hnyls2002** [comment]
Closing this along with the other open unit-test PRs referenced from #20865.

In the AI era, adding unit tests for coverage's sake no longer provides enough value to justify the cost — each PR in this batch consumes reviewer bandwidth and CI time out of proportion to the coverage it adds. The tracking issue itself is already closed, and we are no longer accepting coverage-driven UT PRs from it.

Thanks for the effort — test contributions tied to a concrete bug fix or behavior change are still welcome.
